src/setup/rgw.py

```diff
@@ -14,77 +14,220 @@
 # For any questions about this software or licensing,
 # please email opensource@seagate.com or cortx-questions@seagate.com.
 
+import os
+import time
+import uuid
+import errno
+import shutil
+import socket
+from cortx.utils.validator.v_pkg import PkgV
+from cortx.utils.conf_store import Conf, MappedConf
+from cortx.utils.conf_store.error import ConfError
+from cortx.utils.process import SimpleProcess
+from cortx.utils.log import Log
+from src.setup.error import RgwSetupError
+from src.const import (
+    cortx_rpms, ceph_rpms, ceph_conf_tmpl_file, ceph_conf_tmpl_url,
+    ceph_conf_file, ceph_conf_path, ceph_admin_keyring_file)
+
 
 class Rgw:
     """Represents RGW and Performs setup related actions."""
 
+    _machine_id = Conf.machine_id
+
     @staticmethod
     def validate(phase: str):
-        """Perform validtions."""
+        """Perform validations."""
+
+        Log.info(f'validations started for {phase} phase.')
+
+        if phase == 'post_install':
+            # Perform RPM validations
+            for rpms in [cortx_rpms, ceph_rpms]:
+                PkgV().validate('rpms', rpms)
+            Log.info(f'All required rpms are installed on {Rgw._machine_id} node.')
+        elif phase == 'prepare':
+            Rgw._file_exist(ceph_conf_tmpl_file)
+        elif phase == 'config':
+            Rgw._file_exist(ceph_conf_file)
+
+        Log.info(f'validations completed for {phase} phase.')
 
-        # Perform RPM validations
         return 0
 
     @staticmethod
-    def post_install(config_path: str):
-        """
-        Performs post install operations.
-        """
+    def post_install(cortx_gconf: MappedConf):
+        """Performs post install operations."""
 
+        Log.info('PostInstall phase completed.')
         return 0
 
     @staticmethod
-    def prepare(config_path: str):
-        """
-        Performs prepare phase operations.
-        """
+    def prepare(cortx_gconf: MappedConf):
+        """Perform prepare operations."""
+
+        Log.info('Prepare phase started.')
+
+        try:
+            hostname = cortx_gconf.get(f'node>{Rgw._machine_id}>hostname')
+            host_ip = socket.gethostbyname(hostname)
+            ceph_tmpl_idx = 'ceph_conf_tmpl'
+            # Load template ceph.conf file and updated required field.
+            kv_list = [
+                ('global>fsid', str(uuid.uuid1())),
+                ('global>mon host', host_ip)]
+            Rgw._update_config(ceph_tmpl_idx, ceph_conf_tmpl_url, kv_list)
```

Comment:
Why do we want to update tmpl ? 
Can you use Conf.copy() instead ?

Reply:
Conf.copy() is not working for ini format.
It works only if same section are present in both sorce and destination file, but it is not working if we try to copy source file into some other empty file.

Reply:
New patch is checked in and INI is fixed. Please use it now. 
Refer to examples at https://github.com/Seagate/cortx-utils/blob/main/py-utils/test/conf_store/test_conf_store_ini.py

Reply:
Yes, using Copf.copy() in latest commit.